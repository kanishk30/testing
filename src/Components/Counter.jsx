import React, { useState } from 'react';

function Counter() {
    const [count, setCount] = useState(0);

    const increment = () => {
        setCount(count + 1)
    }

    const decrement = () => {
        setCount(count - 1)
    }

    return <>
        <h3>Counter</h3>
        <button onClick={decrement}> - </button>
        <h3>Count is {count}</h3>
        <button onClick={increment}> + </button>
    </>
}

export default Counter