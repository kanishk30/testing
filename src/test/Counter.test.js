/**
 * 3 step process on hgh level
 * - render the component
 * - select the part that i wnat to checl
 * - compare
 */

import { render, screen, fireEvent } from '@testing-library/react';
import Counter from '../Components/Counter';

describe('Counter test cases', () => {
    test('Checking initial state render', () => {
        // rendering the component
        render(<Counter />);
        // selection.
        const minusText = screen.getByText('-')
        const plusText = screen.getByText('+')
        const countText = screen.getByText('Count is 0')

        // verifying stage...
        expect(minusText).toBeInTheDocument()
        expect(plusText).toBeInTheDocument()
        expect(countText).toBeInTheDocument()
    })

    // increment by 1.
    // mock the click event. fire event by code only.

    test('increment by 1', () => {
        render(<Counter />);
        const plusText = screen.getByText('+');
        fireEvent.click(plusText);

        const isOnePresent = screen.getByText('Count is 1');
        expect(isOnePresent).toBeInTheDocument()
    })

    test('decrement by 5', () => {
        render(<Counter />);
        const minusText = screen.getByText('-');
        fireEvent.click(minusText);
        fireEvent.click(minusText);
        fireEvent.click(minusText);
        fireEvent.click(minusText);
        fireEvent.click(minusText);
        const isOnePresent = screen.getByText('Count is -5');
        expect(isOnePresent).toBeInTheDocument()
    })

    test('Snapshot for counter', () => {
        const { asFragment } = render(<Counter />)
        expect(asFragment()).toMatchSnapshot(); //
    })
})

