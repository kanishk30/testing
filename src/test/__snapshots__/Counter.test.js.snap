// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter test cases Snapshot for counter 1`] = `
<DocumentFragment>
  <h3>
    Counter
  </h3>
  <button>
     - 
  </button>
  <h3>
    Count is 0
  </h3>
  <button>
     + 
  </button>
</DocumentFragment>
`;
